### ISSUE TITLE
Clicking the refresh button resets a variable that has been typed in

### ISSUE THREAD

**Issue by drorm** [metabase/metabase#4905]
### Bugs
If you're filing an issue about a bug please include as much information
as you can including the following.

- Your browser and the version: Chrome  58.0.3029.81 (64-bit) and 52.0.2 (64-bit)
- Your operating system: Linux
- Metabase version: v0.23.0 Built on 2017-03-21
- Metabase hosting environment: Linux

*Repeatable steps to reproduce the issue*
* Load into the page a question that takes a variable with a default. In my case, country: Kenya
* Type in a different value. In my case: Ghana
* Click on refresh. The value is reset. In my case the string goes back to Kenya and the query does use that as the variable.

**Comment by kdoh** [comment]
Hey there @drorm.  I'm having a bit of trouble reproducing this. I tried to follow the steps you outlined by creating a SQL question (I'm assuming you're talking about SQL questions) using our Sample Dataset and when changing a text filter on the state field for PeopleI see the button switch to Run Query as expected and the result reflects the new value.

If you're able, would you mind creating a new question with the following SQL and a default value seeing if you encounter the same issue? This might help us narrow it down to see if it's something specific about your situation.

```
SELECT "PUBLIC"."PEOPLE"."ZIP" AS "ZIP", "PUBLIC"."PEOPLE"."STATE" AS "STATE", "PUBLIC"."PEOPLE"."SOURCE" AS "SOURCE", "PUBLIC"."PEOPLE"."PASSWORD" AS "PASSWORD", "PUBLIC"."PEOPLE"."NAME" AS "NAME", "PUBLIC"."PEOPLE"."LONGITUDE" AS "LONGITUDE", "PUBLIC"."PEOPLE"."LATITUDE" AS "LATITUDE", "PUBLIC"."PEOPLE"."ID" AS "ID", "PUBLIC"."PEOPLE"."EMAIL" AS "EMAIL", "PUBLIC"."PEOPLE"."CREATED_AT" AS "CREATED_AT", "PUBLIC"."PEOPLE"."CITY" AS "CITY", "PUBLIC"."PEOPLE"."BIRTH_DATE" AS "BIRTH_DATE", "PUBLIC"."PEOPLE"."ADDRESS" AS "ADDRESS"
FROM "PUBLIC"."PEOPLE"
WHERE "PUBLIC"."PEOPLE"."STATE" = {{STATE}}
LIMIT 10
```

**Comment by drorm** [comment]
I think the issue is that if you hit enter after typing the name, it works fine, but if you tab or just directly click on the button it reverts.

**Comment by mazameli** [comment]
I believe this was fixed in 0.25. You no longer have to hit enter to commit the filter; clicking `Refresh` should commit the new filter input. I verified this, but are you still experiencing this problem, @drorm?

**Comment by drorm** [comment]
Yes, works fine in 0.25. Thanks.
